EESSI-pilot-install-software.sh

```diff
@@ -138,14 +138,29 @@ echo ">> Setting up \$MODULEPATH..."
 module --force purge
 # ignore current $MODULEPATH entirely
 module unuse $MODULEPATH
-module use $EASYBUILD_INSTALLPATH/modules/all
+if [[ -z "$EESSI_REBUILD_PATH" ]]; then
+    # Default behaviour is to do a full build
+    module use $EASYBUILD_INSTALLPATH/modules/all
+else
+    # If EESSI_REBUILD_PATH is set, we assume software is built and we are just rebuilding the module tree
+    export EASYBUILD_INSTALLPATH_MODULES=$EESSI_REBUILD_PATH/modules
+    [[ -z "$EESSI_REBUILD_MODULEPATH" ]] && module use $EESSI_REBUILD_PATH/modules/all || module use $EESSI_REBUILD_MODULEPATH
+    # Use preferred MNS
+    [[ -z "$EESSI_REBUILD_MNS" ]] && export EASYBUILD_MODULE_NAMING_SCHEME=EasyBuildMNS || export EASYBUILD_MODULE_NAMING_SCHEME=$EESSI_REBUILD_MNS
+    # Only rebuild modules
+    export EASYBUILD_MODULE_ONLY=1
+    # Make sure we can write our lock files
+    export EASYBUILD_LOCKS_DIR=$EESSI_REBUILD_PATH/software
+fi
+
 if [[ -z ${MODULEPATH} ]]; then
     error "Failed to set up \$MODULEPATH?!"
 else
     echo_green ">> MODULEPATH set up: ${MODULEPATH}"
 fi
 
-echo ">> Checking for EasyBuild module..."
+REQ_EB_VERSION='4.3.4'
+echo ">> Checking for EasyBuild module (required version is ${REQ_EB_VERSION})..."
```

Comment:
To be honest since a specific version is required we should be trying to load that version, not the default. If we fail, we should get eb to install the latest release, and if that is too recent use the latest version to install the required version.

Reply:
We can do that in this block with `module avail easybuild/${REQ_EB_VERSION}`, then using a temporary pip installation to _either_ give us the required version _or_ the latest release if that is not available.

Reply:
https://github.com/EESSI/software-layer/issues/89 this is probably also relevant here

Reply:
@bedroge You are right, I will fix this and use `module load` instead